components/wifi/esp8266-driver/ESP8266Interface.cpp

```diff
@@ -505,9 +505,11 @@ int ESP8266Interface::socket_send(void *handle, const void *data, unsigned size)
     } while ((sendStartTime - rtos::Kernel::get_ms_count() < 50)
             && (status != NSAPI_ERROR_OK));
 
-    if (status == NSAPI_ERROR_WOULD_BLOCK) {
-        debug("Enqueuing the event call");
+    if (status == NSAPI_ERROR_WOULD_BLOCK && socket->proto == NSAPI_TCP) {
+        tr_debug("ESP8266Interface::socket_send(): enqueuing the event call");
         _global_event_queue->call_in(100, callback(this, &ESP8266Interface::event));
+    } else if (status == NSAPI_ERROR_WOULD_BLOCK && socket->proto == NSAPI_UDP) {
+        status = NSAPI_ERROR_DEVICE_ERROR;
```

Comment:
Why do we need to modify the NSAPI_ERROR_WOULD_BLOCK to become NSAPI_ERROR_DEVICE_ERROR?

Reply:
I think because UDP sending should never fail.